src/Bridge/Symfony/Bundle/Test/Response.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\Test;
+
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Exception\ClientException;
+use Symfony\Component\BrowserKit\Response as BrowserKitResponse;
+use Symfony\Component\HttpClient\Exception\JsonException;
+use Symfony\Component\HttpFoundation\Response as HttpFoundationResponse;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * HTTP Response.
+ *
+ * @internal
+ *
+ * Partially copied from \Symfony\Component\HttpClient\Response\ResponseTrait
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Response implements ResponseInterface
+{
+    private $httpFoundationResponse;
+    private $browserKitResponse;
+    private $info;
+    private $content;
+    private $jsonData;
+
+    public function __construct(HttpFoundationResponse $httpFoundationResponse, BrowserKitResponse $browserKitResponse, array $info)
+    {
+        $this->httpFoundationResponse = $httpFoundationResponse;
+        $this->browserKitResponse = $browserKitResponse;
+
+        $this->headers = $httpFoundationResponse->headers->all();
+        $this->content = $httpFoundationResponse->getContent();
+        $this->info = $info + [
+            'http_code' => $httpFoundationResponse->getStatusCode(),
+            'error' => null,
+            'raw_headers' => $this->headers,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInfo(string $type = null)
+    {
+        if ($type) {
+            return $this->info[$type] ?? null;
+        }
+
+        return $this->info;
+    }
+
+    /**
+     * Checks the status, and try to extract message if appropriate.
+     */
+    private function checkStatusCode()
+    {
+        $message = 'An error '.$this->info['http_code'].' occured.';
+        if (isset($this->headers['content-type'][0]) && false !== preg_match('#^application/(?:.+\+)?json#', $this->headers['content-type'][0])) {
```

Comment:
what's going on here?
if the regex does not match the content type then it must be an error?
but doesn't `application/problem+json` match?

Reply:
IMO we should avoid doing naive regex matching of MIME types...

Reply:
@bendavies we just test if the body is json in case of error. If it's not, we do nothing.